Lib/ctypes/util.py

```diff
@@ -94,10 +94,13 @@ def find_library(name):
     import re, tempfile
 
     def _is_elf(filename):
-        "Return True if the given file is an ELF file"
+        "Return True if the given file is readable and is an ELF file"
         elf_header = b'\x7fELF'
-        with open(filename, 'br') as thefile:
-            return thefile.read(4) == elf_header
+        try:
+            with open(filename, 'br') as thefile:
+                return thefile.read(4) == elf_header
+        except:
```

Comment:
Can you make the except clause specific?  This would catch a memory error, a typo, keyboard interrupt, etc.